discovery-provider/plugins/notifications/src/processNotifications/mappers/trendingPlaylist.ts

```diff
@@ -0,0 +1,132 @@
+import { Knex } from 'knex'
+import { NotificationRow, PlaylistRow, UserRow } from '../../types/dn'
+import { TrendingPlaylistNotification } from '../../types/notifications'
+import { BaseNotification, Device } from './base'
+import { sendPushNotification } from '../../sns'
+import { ResourceIds, Resources } from '../../email/notifications/renderEmail'
+import { EntityType } from '../../email/notifications/types'
+import { getNumberSuffix } from '../../utils/format'
+
+type TrendingPlaylistNotificationRow = Omit<NotificationRow, 'data'> & {
+  data: TrendingPlaylistNotification
+}
+
+export class TrendingPlaylist extends BaseNotification<TrendingPlaylistNotificationRow> {
+  receiverUserId: number
+  playlistId: number
+  rank: number
+  genre: string
+  timeRange: string
+
+  constructor(
+    dnDB: Knex,
+    identityDB: Knex,
+    notification: TrendingPlaylistNotificationRow
+  ) {
+    super(dnDB, identityDB, notification)
+    const userIds: number[] = this.notification.user_ids!
+    this.receiverUserId = userIds[0]
+    this.playlistId = this.notification.data.playlist_id
+    this.rank = this.notification.data.rank
+    this.genre = this.notification.data.genre
+    this.timeRange = this.notification.data.time_range
+  }
+
+  async pushNotification() {
+    const res: Array<{
+      user_id: number
+      name: string
+      is_deactivated: boolean
+    }> = await this.dnDB
+      .select('user_id', 'name', 'is_deactivated')
+      .from<UserRow>('users')
+      .where('is_current', true)
+      .whereIn('user_id', [this.receiverUserId])
+    const users = res.reduce((acc, user) => {
+      acc[user.user_id] = {
+        name: user.name,
+        isDeactivated: user.is_deactivated
+      }
+      return acc
+    }, {} as Record<number, { name: string; isDeactivated: boolean }>)
+
+    if (users?.[this.receiverUserId]?.isDeactivated) {
+      return
+    }
+
+    const playlistRes: Array<{ playlist_id: number; playlist_name: string }> =
+      await this.dnDB
+        .select('playlist_id', 'playlist_name')
+        .from<PlaylistRow>('playlists')
+        .where('is_current', true)
+        .whereIn('playlist_id', [this.playlistId])
+
+    const playlists = playlistRes.reduce((acc, playlist) => {
+      const { playlist_id, playlist_name } = playlist
```

Comment:
nit - can we move to camel case?

Reply:
so for elements of a destructured object what should i do, rename them inline so it's like 
```
const {playlist_id: playlistId, playlist_name: playlistName} = playlist
```

at least on client side, when we come across an object with snake_case we just destructure and dont rename

Reply:
ah okay good point, no need to rename then